src/Compilers/Core/Portable/Syntax/AbstractWarningStateMap.cs

```diff
@@ -2,13 +2,16 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
+
 using System;
 using System.Collections.Immutable;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Microsoft.CodeAnalysis.Syntax
 {
-    internal abstract class AbstractWarningStateMap<WarningState>
+    internal abstract class AbstractWarningStateMap<TWarningState>
```

Comment:
Can we constrain `TWarningState` to `struct`?

Reply:
Yep that works.